src/index.js

```diff
@@ -1,51 +1,132 @@
+ * Checks if the call expression is a beforeEach call.
+ * @param {object} callee Babel node expression path
+ * @returns {Boolean}
+ */
+function isBeforeEach(callee) {
+  return callee.property && callee.property.name === 'beforeEach';
+}
 
-    if (!existingMetadataDeclaration) {
-      functionBlockBody.unshiftContainer('body', testMetadataAssignment);
-      functionBlockBody.unshiftContainer('body', testMetadataVarDeclaration);
+/**
+ * Gets any existing beforeEach call as a node path.
+ * @param {array} callsArray Babel array of call expression node paths representing calls within a function block
+ * @returns {object} Babel node path call expression of a beforeEach call
+ */
+function getExistingBeforeEach(callsArray) {
+  for (let i = 0; i < callsArray.length; i++) {
+    if (isBeforeEach(callsArray[i].node.expression.callee)) {
+      return callsArray[i].get('expression');
     }
-  } else {
-    const beforeEachFunc = t.functionExpression(
-      null,
-      [],
-      t.blockStatement([testMetadataVarDeclaration, testMetadataAssignment])
-    );
-    const beforeEachExpression = t.expressionStatement(
-      t.callExpression(
-        t.memberExpression(t.identifier('hooks'), t.identifier('beforeEach')),
-        [beforeEachFunc]
-      )
+  }
+}
+
+/**
+ * Adds test metadata statements to the top of an existing beforeEach call function block.
+ * Updates state.opts.transformedModules [] with the name of the test module transformed.
+ * @param {object} state - Babel state
+ * @param {object} beforeEachExpression - the beforeEach Babel call expression
+ * @param {object} t Babel types
+ */
+function addMetaDataToBeforeEach(state, beforeEachExpression, t) {
+  const testMetadataVarDeclaration = getTestMetadataDeclaration(state, t);
+  const testMetadataAssignment = getTestMetadataAssignment(state, t);
+
+  const functionBlock = beforeEachExpression.get('arguments')[0];
+  const functionBlockBody = functionBlock.get('body');
+  const functionBlockBodyStatementsArray = functionBlockBody.get('body');
+  let existingMetadataDeclaration;
+
+  if (functionBlockBodyStatementsArray.length > 0) {
+    existingMetadataDeclaration = functionBlockBodyStatementsArray.find(
+      (node) => hasMetadataDeclaration(node, t)
     );
+  }
+
+  if (!existingMetadataDeclaration) {
+    functionBlockBody.unshiftContainer('body', testMetadataAssignment);
+    functionBlockBody.unshiftContainer('body', testMetadataVarDeclaration);
 
-    babelPath.insertBefore(beforeEachExpression);
+    state.opts.transformedModules.push(state.opts.moduleName);
   }
 }
 
+/**
+ * Creates a new beforeEach call with the test metadata statements.
+ * Inserts this new beforeEach below any existing setup calls, else
+ * at the top of the test module function block.
+ * Updates state.opts.transformedModules [] with the name of the test module transformed.
+ * @param {object} state - Babel state
+ * @param {object} t  - Babel types
+ */
+function writeNewBeforeEach(state, t) {
```

Comment:
`insertBeforeEach` feels more idiomatic to me in terms of naming for babel type stuff. WDYT?

Reply:
I agree. thank you